### ISSUE TITLE
Crew Task Manager

### ISSUE THREAD

**Issue by alexanderson1993** [Thorium-Sim/thorium#1215]
## Detailed Description

From James Porter

This would be a widget that you can place at each station that is one of the pop out windows like composer or messaging.  The purpose is to assign tasks to crew members that allows the to progressively complete those tasks rather than using messaging or print outs to give assignments.

### Station Widget
![sv-5uaq_fya6b1huzczd2kw](https://user-images.githubusercontent.com/6558157/44858224-c1851e80-ac2e-11e8-96c3-db59a0515f13.png)

A pull down list at the top will allow them to select one of the tasks they have been assigned.  When it is selected the steps of the task will be displayed in the gray field. As they complete the step they click the “Step Completed” button and they are advanced to the next step.  The arrows at the bottom let them review steps they have already completed. When the last step is completed a notification will be sent to the core widget to “Verify Completion”. If it is verified the words “Completed” are added to the end of the task name, but it isn’t removed from the pull down menu.  Perhaps the text could be grayed out.

Having them register that a step is completed rather than monitoring what step they are looking at will help emphasize a progression and it will also let the first officer know more accurately what step they are on in case they work on different tasks.

### Core Layout

![s2goyy0rot1bqtd5oes6wsa](https://user-images.githubusercontent.com/6558157/44858202-b6ca8980-ac2e-11e8-8496-0b0edc70a406.png)

Staff will first Select the Station they are wanting to assign a task to.  Then they can either upload a text document or select a mission report. The details will be provided in the preview box including the name of the report.  If all is correct they can assign the task.

The format of the reports will be just like damage reports for ease of use.

In the bottom portion of the core layout is where staff can review the tasks and quickly see the progress of the different tasks just like the damage control screen.  If they click on an item in the left field the current steps and individual instructions will appear in the fields on the right.

If we wanted to make the layout a little simpler we could remove the arrows and just have the full text of the report be in that text field rather than the individual step.  I think being able to look at the individual step saves review time so it is a matter of space.

### Command Monitoring

![szl1bt3g16hlmra8jzzbefa](https://user-images.githubusercontent.com/6558157/44858247-ccd84a00-ac2e-11e8-89d1-992e113b2953.png)

Just like the core layout the command can see what every station is working on and their progress.  They can’t see the individual steps, but they can select an item and prioritize it. They can be assigned tasks just like any other station.  If it is easier to make this two different widgets that makes sense.

Prioritize - Perhaps that tasks gets highlighted on the first officer station and a message is sent to that officer saying, “The #station has requested that you prioritize your work on #task.”  Perhaps if it can have a way of indicating that it is prioritized in the station widgets pull down list that would help add emphasis.

### Mission/Ship Configurations

It would be nice to have a master document (one file) we can create and upload to a simulator configuration as our default tasks that can be assigned.  If “Step1:” can delineate each step can have we have the system look for another clue like parentheses “(Quartermaster report 342323)” to indicate that the next set of steps are for another task?

Certain missions have tasks that are connected with them.  We could store all of these externally, but it would be nice that if we start up a mission then these specific ones are added to the list of possible tasks or we could perhaps add them to the timeline.

### Future consideration

What if Damage Control had the ability to create tasks.  It would reduce the interactions of the damage control officer, but would make it so this can become really robust.  So if they need a call made they can create a task to do so. It would also help them know when things are actually done and what steps they are waiting on.
